packages/compass-preferences-model/src/preferences.ts

```diff
@@ -917,30 +788,30 @@ export class Preferences {
     attributes: Partial<UserPreferences> = {}
   ): Promise<AllPreferences> {
     const keys = Object.keys(attributes) as (keyof UserPreferences)[];
-    const originalPreferences = await this.fetchPreferences();
+    const originalPreferences = this.getPreferences();
```

Comment:
Just to mention it explicitly and make sure it’s intentional, this means that if a user has two Compass instances running and changes preferences first in one and then the other, now that second write always undoes the first one, rather than merging the updates

Reply:
Oh! This seems like a breaking change i am introducing. I think I need to align this and avoid any unwanted behaviour.

Reply:
While going through the changes, I stumbled on [this page](https://www.electronjs.org/docs/latest/tutorial/process-model#the-main-process) and realized that electron has a single main process and all the browser windows share it. So any change made in any Compass window should reflect on other and should not be out of sync. Plus I also tested it and works just fine. Let me know what you think about it.

Reply:
Two instances of the app or two windows in the same app?

Reply:
multiple windows within the same app.

Reply:
@mabaasit Right, that’s how it works on macOS or when you use the “New Connection” button. On other OSes, starting Compass multiple times through the regular mechanism for doing so will create actual different Compass main processes.

Again, no particularly strong feelings here and this is still a change in behavior that we can make, we should just be intentional about it.

Reply:
added [2e12f02](https://github.com/mongodb-js/compass/pull/4600/commits/2e12f020dfece0322a29a6d91be9eed47e30ec25) to address this. 

However, when we have two different compass instances and user updates any preferences it will not be reflected in other one unless some user action triggers fetching of preferences (which is the current behaviour).